vioserial/sys/Control.c

```diff
@@ -55,8 +56,13 @@ VIOSerialSendCtrlMsg(
         while(!virtqueue_get_buf(vq, &len))
         {
             LARGE_INTEGER interval;
-            interval.QuadPart = -1;
+            interval.QuadPart = -100000;
```

Comment:
> For me 10 milliseconds wait time is too small.
We always want for the host to release the memory it is using.
I understand some corner cases can happen, definitely when we don't have queue reset mechanism that should have being used here.

I suppose it is OK to increase the time interval from 10 ms to something larger (e.g. a second). The time limit for power down is AFAIK not that strict. The initial value of `1` (100 ns) seemed to me quite unreaslistic for passive waiting, thus, I increased it.

Of course, it is possible to implement more complex scheme of dynamically adjusting the waiting time.

> Also, why does this happen with control queue at all? I can understand data queue (back end was disconnected for example).

That I am not sure. I got crashed VM and kernel memory dump clearly indicating the serial port device not reacting quickly enough for system shutdown. I do not know what exactly was happening on the host. However, I think it would be nice if the guest was able to shutdown itself properly even if the host gets into a trouble.

I am unable to reproduce the issue (it appeared once during our automatic testing), although it may, of course happen in the future giving me another opportunity to investigate.

Reply:
Ad 1)
I increased the timeout. Actually, there are now two time intervals. At first, `vioser` waits for 100 ms for the host. If that is not sufficient, 1-second waits are performed. In case of a shutdown/restart/powerdown, `vioser` waits for 100+1000 ms. I suppose I can still prolong the wait in the latter case if necessary.

Ad 2)
I can see nothing special in the host logs. Should I look somewhere specifically for this information?